c++/include/orc/Writer.hh

```diff
@@ -0,0 +1,294 @@
+     */
+    bool getBlockPadding() const;
+
+    /**
+     * Set Run length encoding version
+     */
+    WriterOptions& setRleVersion(RleVersion version);
+
+    /**
+     * Get Run Length Encoding version
+     */
+    RleVersion getRleVersion() const;
+
+    /**
+     * Set compression kind.
+     */
+    WriterOptions& setCompression(CompressionKind comp);
+
+    /**
+     * Get the compression kind.
+     * @return if not set, return default value which is ZLIB.
+     */
+    CompressionKind getCompression() const;
+
+    /**
+     * Set the encoding strategy.
+     */
+    WriterOptions& setEncodingStrategy(EncodingStrategy strategy);
+
+    /**
+     * Get the encoding strategy.
+     * @return if not set, return default value which is SPEED.
+     */
+    EncodingStrategy getEncodingStrategy() const;
+
+    /**
+     * Set the compression strategy.
+     */
+    WriterOptions& setCompressionStrategy(CompressionStrategy strategy);
+
+    /**
+     * Get the compression strategy.
+     * @return if not set, return default value which is speed.
+     */
+    CompressionStrategy getCompressionStrategy() const;
+
+    /**
+     * Set the writer version.
+     */
+    WriterOptions& setWriterVersion(WriterVersion version);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default value which is WriterVersion_ORIGINAL
+     */
+    WriterVersion getWriterVersion() const;
+
+    /**
+     * Set the padding tolerance.
+     */
+    WriterOptions& setPaddingTolerance(double tolerance);
+
+    /**
+     * Get the padding tolerance.
+     * @return if not set, return default value which is zero.
+     */
+    double getPaddingTolerance() const;
+
+    /**
+     * Set the memory pool.
+     */
+    WriterOptions& setMemoryPool(MemoryPool * memoryPool);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default memory pool.
+     */
+    MemoryPool * getMemoryPool() const;
+
+    /**
+     * Set the error stream.
+     */
+    WriterOptions& setErrorStream(std::ostream& errStream);
+
+    /**
+     * Get the error stream.
+     * @return if not set, return std::err.
+     */
+    std::ostream * getErrorStream() const;
+
+    /**
+     * Set whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     */
+    WriterOptions& setEnableStats(bool enable);
+
+    /**
+     * Get whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     * @return if not set, the default is true
+     */
+     bool getEnableStats() const;
+
+    /**
+     * Set whether or not to update string min/max stats
+     */
+    WriterOptions& setEnableStrStatsCmp(bool enable);
+
+    /**
+     * Get whether or not to update string min/max stats
+     * @return if not set, the default is false
+     */
+    bool getEnableStrStatsCmp() const;
+
+    /**
+     * Set whether or not to write row group index
+     */
+    WriterOptions& setEnableIndex(bool enable);
```

Comment:
We don't want lots of variants for users to write the files with. Most users take the default and the others can either choose to completely disable them or increase the index stride. Both of which save space, but reduce the effectiveness of predicate push down.